### PR TITLE
Create Use-Case Interactors + Presenters with Backend for join/create/list games

### PR DESCRIPTION
**Pull request by klokailo**
- Created use case interactors for:
  - Listing rooms
    - Sends a GET request to `/list-games` using our web client
    - Updates the list games view using its presenter
  - Creating and joining games
    - Sends a POST request to `/create-game` or `/join-game` using our web client
    - Data for the request comes from the welcome view model
    - Once it has received a response, either establishes a websocket and switches to the game room view, or gives an error message

Fully tested with the backend! All the buttons work.

### CODE REVIEW

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java
Comment (xinlei55555) [inline frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java:1]: I am not sure if we want an array here, as the size couldn't be changed afterwards, and I am guessing we want the number of rooms to not be fixed

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewController.java
Comment (xinlei55555) [inline frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewController.java:1]: You mentioned in the past that we should use the constructor instead, and declare the @Autowired through the constructor, should we do this, or is declaring each attribute separately in the class also okay?

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewModel.java
Comment (xinlei55555) [inline frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewModel.java:1]: Good idea for the default value, or else it just printed "Enter Your Name".

**Review by xinlei55555 — CHANGES_REQUESTED** [review]
A few elements of uncertainty

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java
Comment (klokailo) [reply frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java:1]: What is stored in the view model is just the actually array of lines that we write in the text field. This is following clean architecture.
In the case that more rooms are created, we simply can simply poll the `/list-games` endpoint and create a new array using the `List<String>` of rooms it returns.

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewController.java
Comment (klokailo) [reply frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/welcome/WelcomeViewController.java:1]: Yes, I can change these to use constructor dependency injection.

File: frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java
Comment (xinlei55555) [reply frontend/src/main/java/dev/totallyspies/spydle/frontend/interface_adapters/list_rooms/ListRoomsViewModel.java:1]: Good to know :+1:

**Review by xinlei55555 — APPROVED** [review]

**Pull request merged**